src/flows/pipes/Schedule/History.tsx

```diff
@@ -0,0 +1,293 @@
+import React, {FC, useEffect, useState, useRef, useContext} from 'react'
+import {
+  getTask,
+  patchTask,
+  getTasksRuns,
+  getTasksRun,
+  postTasksRun,
+  getTasksRunsLogs,
+} from 'src/client/generatedRoutes'
+import {
+  EmptyState,
+  Button,
+  ComponentSize,
+  ComponentColor,
+  SlideToggle,
+  IndexList,
+  DapperScrollbars,
+  Overlay,
+} from '@influxdata/clockface'
+import {createDateTimeFormatter} from 'src/utils/datetime/formatters'
+import {PopupContext} from 'src/flows/context/popup'
+
+import {event} from 'src/cloud/utils/reporting'
+
+interface Props {
+  task: any
+}
+
+interface LogProps {
+  taskID: string
+  runID: string
+}
+
+interface RunProps {
+  task: any
+  run: any
+}
+
+const duration = (start: string, finish: string): string => {
+  let timeTag = 'seconds'
+  const _start = Date.parse(start)
+  const _finish = Date.parse(finish)
+
+  if (isNaN(_finish) || isNaN(_start)) {
+    return ''
+  }
+
+  let diff = (_finish - _start) / 1000
+
+  if (diff > 60) {
+    diff = Math.round(diff / 60)
+    timeTag = 'minutes'
+  }
+
+  return `${diff} ${timeTag}`
+}
+
+const RunLogs: FC<LogProps> = ({taskID, runID}) => {
+  const {closeFn} = useContext(PopupContext)
+  const [logs, setLogs] = useState<any[]>([])
+
+  const formatter = createDateTimeFormatter('YYYY-MM-DD HH:mm:ss')
+
+  useEffect(() => {
+    getTasksRunsLogs({taskID, runID}).then(resp => {
+      if (resp.status !== 200) {
+        throw new Error(resp.data.message)
+      }
```

Comment:
where is this being handled?

Reply:
at the nearest error boundary. i dont understand

Reply:
Not sure if that's scoped to the pipe or not, but I've seen things collapse too often on a faulty configuration in notebooks that render the notebook inaccessible. Would love it if we could handle these so that we don't end in a potentially inoperable state

Reply:
k. wrapped the whole component in an error boundary to contain errors into the sidebar. lemme know if that's what you were thinking